### DISCUSSION TITLE
How to handle governing equations with if-else conditions

### DISCUSSION THREAD

**Discussion by kaushikb258** [do-mpc/do-mpc#167] [Q&A]
I have some model params in the governing equations that take different values in different regimes. For example, using the CSTR example, suppose if the reaction rates are different for temperature T_R < 1000 and T_R > 1000, how to code this? I tried using min(_, 0) and max(_, 0) but Casadi throws runtime errors. So my question is how does one handle governing equations when the parameters in them take different values in different regimes?

**Comment by 4flixt** [comment]
Hi @kaushikb258,

while we do not recommend it, CasADi does support [logic operations](https://web.casadi.org/docs/#list-of-operations). See for example

```
fmin(a,b)
```

which works similar to ``np.minimum``. You can use these operations in the same way you would use a ``sin`` function or any other operation. The problem is just that their gradient is not continuous which will often lead to problems with our preferred optimizer (ipopt). I suggest you try using it but don't expect that it will work flawlessly.

Best
@4flixt 

P.S.: Sorry for the late reply. I unfortunately overlooked the push notification regarding this issue. Also note that I have converted your issue to a discussion thread.
